### ISSUE TITLE
Cannot find/create custom huey queue

### ISSUE THREAD

**Issue by crimsonpython24** [coleifer/huey#605]
I have this following code:

**tasks.py**
```python
@periodic_task(crontab(minute='*/1'))
def record_racedata(teamname):
    team = nitrotype.Team(teamname)
    team_id = team.data["info"]["teamID"]
    timestamp = datetime.now().timestamp()
    # ...
```

**settings.py**
```python
HUEY = {
    'huey_class': 'huey.RedisHuey',  
    'name': DATABASES['default']['NAME'],  
    'results': True,  
    'store_none': False, 
    'immediate': False,  
    'utc': True,  
    'blocking': True,  
    'connection': {
        'host': 'localhost',
        'port': 6379,
        'db': 0,
        'connection_pool': None, 
        'read_timeout': 1,  
        'url': None,  
    },
    'consumer': {
        'workers': 1,
        'worker_type': 'thread',
        'initial_delay': 0.1,  
        'backoff': 1.15,  
        'max_delay': 10.0, 
        'scheduler_interval': 1,
        'periodic': True, 
        'check_worker_health': True, 
        'health_check_interval': 1, 
    },
}
```

And when I do `python manage.py run_huey` this throws an error because I did not provide the argument `teamname`.

The question is, how do I create a custom queue that runs the tasks with arguments? I only defined these functions in `tasks.py` and then huey runs them automatically. I don't know the right keywords to look up in the documentation either.

In other words, I am looking for something like, if possible (pseudocode):
```python
RUN_HUEY_TASKS = [
    run_task_1(arguments1)
    run_task_1(arguments2)
    run_task_2(arguments1)
    # ...
```

This is more of a usage question but I came here to ask for help as rarely anyone was watching "huey" on Stack Overflow.

Thanks in advance.

**Comment by michalpleszczynski** [comment]
From the note in the docs for [periodic_task](https://huey.readthedocs.io/en/latest/api.html#Huey.periodic_task):

```
Because functions decorated with periodic_task are meant to be executed at intervals in isolation,
they should not take any required parameters nor should they be expected to return a meaningful value.
```

You'll most likely have to either:
- run one `periodic_task` without arguments that will get the `teamname`s from some storage then iterate over them and do whatever needs to be done.
- create one [dynamic periodic task](https://huey.readthedocs.io/en/latest/guide.html#dynamic-periodic-tasks) per `teamname`

**Comment by crimsonpython24** [comment]
Can I assume that dynamic periodic tasks will also automatically execute (when placed in `tests.py`) in a Django project, or do I have to create a custom configuration?

**Comment by michalpleszczynski** [comment]
I don't know, seems like a weird place to define your tasks, you'll have to try it out yourself

**Comment by coleifer** [comment]
You probably want some partial application so you can create a wrapper function which then calls your function w/the given arg.

That said, this is a python programming issue, and not in any way a huey issue. I'd suggest trying StackOverflow for this type of question.

**Comment by crimsonpython24** [comment]
Thanks to @michalpleszczynski. Example usage:

```python
def get_racedata(teamname, cron_min):
    team = nitrotype.Team(teamname)
    team_id = team.data["info"]["teamID"]
    # ...

@task()
def record_racedata(teamname, cron_min):
    def wrapper():
        get_racedata(teamname, cron_min)

    schedule = crontab(minute=cron_min)
    task_name = 'record_racedata_%s_%s' % (int(time.time()), teamname)
    periodic_task(schedule, name=task_name)(wrapper)

record_racedata('teamname', '*/1')
```

**Comment by verydemo** [comment]
> Thanks to @michalpleszczynski. Example usage:
> 
> ```python
> def get_racedata(teamname, cron_min):
>     team = nitrotype.Team(teamname)
>     team_id = team.data["info"]["teamID"]
>     # ...
> 
> @task()
> def record_racedata(teamname, cron_min):
>     def wrapper():
>         get_racedata(teamname, cron_min)
> 
>     schedule = crontab(minute=cron_min)
>     task_name = 'record_racedata_%s_%s' % (int(time.time()), teamname)
>     periodic_task(schedule, name=task_name)(wrapper)
> 
> record_racedata('teamname', '*/1')
> ```
> 
> Took me a while to find out that I can simply call the function within `tasks.py` and huey will automatically handle it.
> 
> Not sure if adding `if __name__ == '__main__'` will still work, but the code above does.



hi, i can delete task by this task_name?  or func periodic_task can Incoming  id？

**Comment by crimsonpython24** [comment]
For the first question, I don't think you can delete a task. If you place a task in `tasks.py`, Huey will automatically run them. You may want to look at "pausing tasks" in Huey, in case you don't want to terminate the task.

For the second question, per my experience, I don't think `periodic_task` can take incoming ID either. That's the problem that I faced in this issue because `periodic_task` will be automatically run. I used `tasks` and call them separately by passing in variables.

```python
@task()
def record_racedata(teamname, cron_min):   # place teamname (or id) and scheduler as params
    def wrapper():
        # main function to execute

    schedule = crontab(minute=cron_min)
    task_name = 'record_racedata_%s_%s' % (int(time.time()), teamname)
    periodic_task(schedule, name=task_name)(wrapper)

record_racedata('teamname', '*/1')   # use custom teamname and scheduler
```

I don't feel like there's much difference by using `task()` but it allows you to pass in parameters like an incoming ID. You can still put everything in `tasks.py` and Huey will automatically detect/run them.

**Comment by verydemo** [comment]
thank you for your reply!
Can Huey pausing periodic_task through task_name? I don't want to save task_id.

**Comment by crimsonpython24** [comment]
That I'm not exactly sure, you probably have to wait for @coleifer's reply or open a new issue.
